### ISSUE TITLE
BBC: Crypt-MagicSignatures-Key-0.21 fails with perl 5.31.5

### ISSUE THREAD

**Issue by eserte** [Perl/perl5#17247]
**Description**
The test suite of Crypt-MagicSignatures-Key-0.21 fails with perl 5.31.5. 5.31.4 and older passes.

Overview of test reports:
http://fast-matrix.cpantesters.org/?dist=Crypt-MagicSignatures-Key-0.21

**Comment by jkeenan** [comment]
Xref: https://rt.cpan.org/Ticket/Display.html?id=130890

**Comment by jkeenan** [comment]
Bisection points to:
```
e8864dba80952684bf3afe83438d4eee0c3939a9 is the first bad commit
commit e8864dba80952684bf3afe83438d4eee0c3939a9
Author: Matt Turner <mattst88@gmail.com>
Date:   Wed Sep 4 21:48:56 2019 -0700

    Clean up U8TO*_LE macro implementations
    
    The code guarded by #ifndef U32_ALIGNMENT_REQUIRED attempts to optimize
    byte-swapping by doing unaligned loads, but accessing data through
    unaligned pointers is undefined behavior in C. Moreover, compilers are
    more than capable of recognizing these open-coded byte-swap patterns and
    emitting a bswap instruction, or an unaligned load instruction, or a
    combined load, etc. There's no need for multiple paths to attain the
    desired result.
    
    See https://rt.perl.org/Ticket/Display.html?id=133495

:100644 100644 77a4c8489676e76d025779a74ba4e14794f05840 02c0baad087e440fcee08beea07e401227e0fc51 M	hv_macro.h
:100644 100644 bd09c2f9389127a343d51a235fabe3ecade9d8a0 5ee879485df2e3b79af2e5176b1c459f1237bdad M	stadtx_hash.h
:100644 100644 c9b60ccb32b4f066470faa8aa90cb5449c0903c8 2fb391a2337ddf33870e6e2ee4d979190240157d M	zaphod32_hash.h
bisect run success
That took 3481 seconds.

commit e8864dba80952684bf3afe83438d4eee0c3939a9
Author:     Matt Turner <mattst88@gmail.com>
AuthorDate: Wed Sep 4 21:48:56 2019 -0700
Commit:     Tony Cook <tony@develop-help.com>
CommitDate: Tue Oct 8 09:27:43 2019 +1100
```
This commit was also identified in https://github.com/Perl/perl5/issues/17244 as causing breakage to CPAN distribution Algorithm-MinPerfHashTwoLevel.

Thank you very much.
Jim Keenan

**Comment by jkeenan** [comment]
This ticket was closed yesterday after commit 2e38307.  However, just now I built blead at HEAD (a6d5b8290f10eb709eb405ee94145c9620f8e5a2) and tried to install Crypt::MagicSignatures::Key using ```cpanm```.  The installation failed.
```
ok 40 - Too large
not ok 41 - Correct subname

#   Failed test 'Correct subname'
#   at t/12-key-gen.t line 142.
#          got: 'Math::Prime::Util::PP::random_nbit_prime'
#     expected: 'Math::Prime::Util::random_nbit_prime'
ok 42 - Correctly overwritten
ok 43 - Key Generation
ok 44 - Maximum rounds
1..44
# Looks like you failed 1 test of 44.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/44 subtests 

Test Summary Report
-------------------
t/12-key-gen.t (Wstat: 256 Tests: 44 Failed: 1)
  Failed test:  41
  Non-zero exit status: 1
```
Re-opening issue.

Thank you very much.
Jim Keenan
